### PR TITLE
Update files

### PR DESCRIPTION
**Pull request by valter-junior**
Back end to review

### CODE REVIEW

File: backend/db.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+const pool = new Pool({
+  user: "postgres",
+  password: "postgres",
+  host: "localhost",
+  port: 5432,
+  database: "todolist"
+});
+
+module.exports = pool;
```
Comment (jfmrm) [inline backend/db.js]: add empty line

File: backend/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const PORT = process.env.PORT || 3001;
+
+//middleware
+
+app.use(cors());
+app.use(express.json());
+
+
+
+
```
Comment (jfmrm) [inline backend/index.js]: remove these lines

File: backend/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+app.post("/sign-in", async (req, res) => {
+  try {
+    const { firstName, lastName, login, password } = req.body;
+    const newTodo = await pool.query("INSERT INTO todo (firstName, lastName, login, password) VALUES($1, $2, $3, $4) RETURNING *", [firstName, lastName, login, password]);
+    res.json(newTodo.rows[0]);
+
+  } catch (err) {
+    console.error(err.message);
```
Comment (jfmrm) [inline backend/index.js]: add a `res.json()` with an error message

File: backend/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    
+  }
+});
+
+
+
+
+app.listen(PORT, () => {
+  console.log(`Server listening on ${PORT}`);
+});
```
Comment (jfmrm) [inline backend/index.js]: add empty line

File: backend/index.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    const todo = await pool.query("SELECT * FROM todo WHERE todo_id = $1", [id]);
+    res.json(todo.rows[0]);
+  } catch (error) {
+    console.error(err.message);
+    
+  }
+});
+
+
+
```
Comment (jfmrm) [inline backend/index.js]: remove lines

File: backend/index.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+app.get("/sign-in", async (req, res) => {
+  try {
+    const allTodo = await pool.query("SELECT * FROM todo");
+    res.json(allTodo.rows);
+  } catch (err) {
+    console.error(err.message);
+    
+  }
+});
```
Comment (jfmrm) [inline backend/index.js]: this route doesn't make sense to me, what should it do?

**Review by jfmrm — CHANGES_REQUESTED** [review]
you have to add node_modules to .gitignore, it should not be tracked on git. From root project folder run:
```
git rm backend/node_modules
git commit -m "Remove node modules"
git push
```
add it to .gitignore.

overall its good enough for now, I'll teach you some software engineering fundamentals so we can refactor this later.

File: backend/index.js (hunk 6 above)
Comment (valter-junior) [reply backend/index.js]: This route list all users that are in database, it's necessary?

Comment (jfmrm) [reply backend/index.js]: don't think so. It's not in the doc you created.

**Review by jfmrm — COMMENTED** [review]
tight

File: backend/index.js (hunk 6 above)
Comment (jfmrm) [reply backend/index.js]: just remove this one and you can merge

Comment (valter-junior) [reply backend/index.js]: ok

Comment (valter-junior) [reply backend/index.js]: wait, so I have one question about the new route with bearer, Can you review?

**Comment by jfmrm** [comment]
> wait, so I have one question about the new route with bearer, Can you review?

@valter-junior I reviewed it, what is your question?

**Comment by valter-junior** [comment]
> > wait, so I have one question about the new route with bearer, Can you review?
> 
> @valter-junior I reviewed it, what is your question?

It's right the route?

File: backend/index.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  try {
+    const { username } = jwt.decode(token, SECRET);
+    if (username === USER) {
+      done(null, username, { scope: 'read' });
+      return;
+    }
+    done(null, false);
+  } catch (error) {
+    done(null, false);
+  }
```
Comment (jfmrm) [inline backend/index.js]: this method is responsible for authenticating, so if something is wrong it is here.

File: backend/DataBase.sql [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    lastName VARCHAR(50) NOT NULL,
+    login VARCHAR(50) UNIQUE NOT NULL,
+    password VARCHAR(50) UNIQUE NOT NULL
+);
+
+CREATE TABLE list(
+    id SERIAL PRIMARY KEY,
+    title VARCHAR(50),
+    description VARCHAR(256)
+);
```
Comment (jfmrm) [inline backend/DataBase.sql]: add empty line

**Review by jfmrm — COMMENTED** [review]
still using passport here

File: backend/.gitignore [hunk 9]
```diff
@@ -0,0 +1,4 @@
+/node_modules
+/models
+/config
+/db.js
```
Comment (jfmrm) [inline backend/.gitignore]: missing empty line at the end.

Comment (jfmrm) [inline backend/.gitignore]: the reason why the gitignore is not working is that u are using absolute paths. try something like this:
```
node_modules/
```

File: backend/config/.sequelizerc [hunk 10]
```diff
@@ -0,0 +1,8 @@
+const path = require('path')
+  
+module.exports = {
+  "config": path.resolve('./server/config', 'config.json'),
+  "models-path": path.resolve('./server/models'),
+  "seeders-path": path.resolve('./server/seeders'),
+  "migrations-path": path.resolve('./server/migrations')
+};
```
Comment (jfmrm) [inline backend/config/.sequelizerc]: missing empty line

File: backend/controllers/controllers.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+const Notes = require('/home/valter/Projects/todolist/backend/models/list.js')
+
+const note = await Notes.create({
+  
+})
```
Comment (jfmrm) [inline backend/controllers/controllers.js]: missing empty line

File: backend/controllers/controllers.js [hunk 12]
```diff
@@ -0,0 +1,5 @@
+const Notes = require('/home/valter/Projects/todolist/backend/models/list.js')
```
Comment (jfmrm) [inline backend/controllers/controllers.js]: using absolute path. should be relative.

File: backend/controllers/controllers.js (hunk 11 above)
Comment (jfmrm) [inline backend/controllers/controllers.js]: what is the purpose of this file?

File: backend/controllers/index.js [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+  }
+};
+
+module.exports = {
+  createPost,
+  getAllNotes,
+  updateNotes,
+  deleteNotes
+
+}
```
Comment (jfmrm) [inline backend/controllers/index.js]: missing empty line.

File: backend/controllers/index.js [hunk 14]
```diff
@@ -0,0 +1,66 @@
+const Notes = require('../models/list');
+
+
+const createPost = async (req, res) => {
+  try {
+    const note = await Notes.create({
+      title: req.body.title,
+      description: req.body.description});
```
Comment (jfmrm) [inline backend/controllers/index.js]:
```suggestion
    const note = await Notes.create({
      title: req.body.title,
      description: req.body.description
    });
```

File: backend/controllers/index.js [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+  } catch (error) {
+    return res.status(500).json({error: error.message})
+  }
+}
+
+const getAllNotes = async (req, res) => {
+  try {
+    const notes = await Notes.findAll({
+      attributes: ['id', 'title', 'description'],
+
```
Comment (jfmrm) [inline backend/controllers/index.js]: no need for this extra line.

File: backend/index.js [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+
+    res.json("User was deleted!");    
+  } catch (error) {
+    console.error(error.message);    
+  }
+});
+
+
+
+
```
Comment (jfmrm) [inline backend/index.js]: no need for these extra lines

File: backend/models/list.js [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+      name: "list_pkey",
+      unique: true,
+      fields: [
+        { name: "id" },
+      ]
+    },
+  ]
+});
+
+module.exports = Notes;
```
Comment (jfmrm) [inline backend/models/list.js]: missing empty line.

File: backend/.gitignore (hunk 9 above)
Comment (valter-junior) [reply backend/.gitignore]: I didn't can  resolve this after the changes.

File: backend/controllers/controllers.js (hunk 11 above)
Comment (valter-junior) [reply backend/controllers/controllers.js]: This was not utilized. I deleted.

File: backend/app.js [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+app.get("/sign-in", controllersUser.getAllUsers);
+
+app.delete("/sign-in/:id", controllersUser.deleteUser);
+
+app.use('/lists', listRouter);
+
+app.post("/task", controllerTask.createTask);
+  
+app.get("/task", controllerList.getAllLists);
```
Comment (jfmrm) [inline backend/app.js]: break down in routers, just like list router

File: backend/app.js [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+      req.user = user;
+      next();
+    });
+  } else {
+    res.sendStatus(401); 
+  }
+}
+
+
+
```
Comment (jfmrm) [inline backend/app.js:47]: remove extra lines

File: backend/app.js [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+    res.json({"msg": "unauthorized"});
+
+  }
+  },
+);
+
+app.post("/sign-in", controllersUser.createUser);
+
+
+
```
Comment (jfmrm) [inline backend/app.js:65]: remove extra lines

File: backend/controllers/controllerList.js [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+const deleteLists = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const deleted = await prisma.list.delete({
+      where: { id: Number(id) }
+    });
+    if (deleted) {
+      return res.status(200).json({"msg": "List deleted"});
```
Comment (jfmrm) [inline backend/controllers/controllerList.js]: deleted status is 204

File: backend/controllers/controllerTask.js [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+const createTask = async (req, res) => {
+  try {
+    const { listid, description} = req.body
+    const list = await prisma.tasks.create({
+      data: {
+        listid,
+        description,
+      }
+    });
+    return res.status(200).json({ list });
```
Comment (jfmrm) [inline backend/controllers/controllerTask.js]: remove enclosing object, return plain object

File: backend/controllers/controllerTask.js [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+    return res.status(500).json({error: error.message})
+  }
+}
+
+const getAllTasks = async (req, res) => {
+  try {
+    
+    const lists = await prisma.tasks.findMany()
+    
+    return res.status(200).json({ lists });
```
Comment (jfmrm) [inline backend/controllers/controllerTask.js:26]: same as above

File: backend/controllers/controllerUser.js [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+        data: {
+          firstname,
+          lastname,
+          email,
+          password
+        },      
+      });
+      return res.status(201).json({
+        user,
+      });
```
Comment (jfmrm) [inline backend/controllers/controllerUser.js:17]: return plain user (remove enclosing object)

File: backend/controllers/controllerUser.js [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+    } catch (error) {
+      return res.status(500).json({error: error.message})
+    }
+  }
+
+const getAllUsers = async (req, res) => {
+    try {
+      const users = await prisma.users.findMany();
+      return res.status(200).json({ users });
```
Comment (jfmrm) [inline backend/controllers/controllerUser.js:28]: same as above

File: backend/controllers/controllerUser.js [hunk 26]
```diff
@@ -0,0 +1,52 @@
...
+const deleteUser = async (req, res) => {
+  try {
+    const  id  = +req.params.id
+    const deleted = await prisma.users.delete({
+      where: {
+        id
+      },
+    })
+    if (deleted) {
+      return res.status(200).json({"msg": "User deleted"});
```
Comment (jfmrm) [inline backend/controllers/controllerUser.js:45]: delete response status 204

**Review by jfmrm — CHANGES_REQUESTED** [review]

File: backend/app.js (hunk 18 above)
Comment (valter-junior) [reply backend/app.js]: Done

File: backend/app.js (hunk 19 above)
Comment (valter-junior) [reply backend/app.js:47]: Done

File: backend/app.js (hunk 20 above)
Comment (valter-junior) [reply backend/app.js:65]: Done

File: backend/controllers/controllerList.js (hunk 21 above)
Comment (valter-junior) [reply backend/controllers/controllerList.js]: Done

File: backend/controllers/controllerTask.js (hunk 22 above)
Comment (valter-junior) [reply backend/controllers/controllerTask.js]: Done

File: backend/controllers/controllerTask.js (hunk 23 above)
Comment (valter-junior) [reply backend/controllers/controllerTask.js:26]: Done

File: backend/controllers/controllerUser.js (hunk 24 above)
Comment (valter-junior) [reply backend/controllers/controllerUser.js:17]: Done

File: backend/controllers/controllerUser.js (hunk 25 above)
Comment (valter-junior) [reply backend/controllers/controllerUser.js:28]: Done

File: backend/controllers/controllerUser.js (hunk 26 above)
Comment (valter-junior) [reply backend/controllers/controllerUser.js:45]: Done

**Pull request merged**